File: JS-MongoDB/insertGenerator.js
var __dataframe = require("dataframe-js").DataFrame;
var __nosql = require("mongodb");
var __fs = require("fs")
var __parse = require("csv-parse");

const func = async () => {

	const __nosqlClient = new __nosql.MongoClient(
		"mongodb://127.0.0.1:27017/",
		{ useUnifiedTopology: true }
	);
	
	await __nosqlClient.connect();
	
	const nosql = __nosqlClient.db("mydb").collection("weather");
	var pathOfCSV = "/home/devcirillo/Documenti/GitHub/nosql-fly-implementation/FLY-MongoDB/weatherHistory.csv"
	/* var weatherCSVClient = containerClient.getBlobClient(pathOfCSV);
	var weatherCSV = await weatherCSVClient.download(); */ // -> Fly code gen.

	const insertWeatherCSV = async () => {
		
		let i = 0;
		let features;
		let objects = [];
		
		await new Promise((resolve) => {
		
			__fs.createReadStream(pathOfCSV)
			.pipe(__parse())
			.on("data", (row) => {
				if(i == 0) {
					features = row;
					++i;
				} else if(i >= 1 && i <= 10) {
					let object = { };
					for([index, value] of features.entries())
						object[features[index]] = row[index];
					objects.push(object);
					++i;
				} else if(i < 1)
					++i;
			})
			.on("end", () => {
				resolve();
			});
		});
		
		return objects;
	}
	
	await nosql.insertMany((await insertWeatherCSV())); // TEST PASSED! -> Insert from csv.

	var statement = "{ \"name\": \"Antonio\", \"surname\": \"Cirillo\", \"age\": 21 }"
	let insertStudent;
	if(statement.charAt(0) === "[")
		insertStudent = JSON.parse(statement);
	else
		insertStudent = JSON.parse("[" + statement + "]");
	
	await nosql.insertMany(insertStudent); // TEST PASSED! -> Insert single object.

	var statement = "[ " + "{ \"name\": \"Antonio\", \"surname\": \"Cirillo\", \"age\": 21 }" + ", " + "{  \"name\": \"Giovanni\", \"surname\": \"Rapa\", \"age\": 21 }" + " ]"
	let insertStudents;
	if(statement.charAt(0) === "[")
		insertStudents = JSON.parse(statement);
	else
		insertStudents = JSON.parse("[" + statement + "]");
	
	await nosql.insertMany(insertStudents); // TEST PASSED ! -> Insert multi objects.

	await __nosqlClient.close(); 

}

func(); 

File: PY-MongoDB/insertGenerator.py
#import os
#import random
#import time
#import math 
import pandas as pd
#import numpy as np
import json
#import urllib.request
#import pymysql
from pymongo import MongoClient

#import azure.functions as func
#from azure.storage.queue import QueueServiceClient

def main() :
	#__environment = 'azure'
	#__queue_service = QueueServiceClient(account_url='https://"${storageName}".queue.core.windows.net', credential='"${storageKey}"')
	#__event = req.get_json()
	#data = __event['data']
	nosqlClient = MongoClient('mongodb://127.0.0.1:27017/') 
	nosqlDatabase = nosqlClient['mydb']
	nosql = nosqlDatabase['weather']
	pathOfCSV =  '/home/devcirillo/Documenti/GitHub/nosql-fly-implementation/FLY-MongoDB/weatherHistory.csv' 
	#if 'http' in pathOfCSV:
	#	weatherCSV = urllib.request.urlopen(urllib.request.Request(pathOfCSV,headers={'Content-Type':'application/x-www-form-urlencoded;charset=utf-8'}))
	#else:
	#	weatherCSV = open(pathOfCSV,'rw')
	insertWeatherCSV = pd.read_csv(pathOfCSV, skiprows = range(1, 100), nrows = (155 - 100)).to_dict('records')
	
	nosql.insert_many(insertWeatherCSV)
	
	statement =  '[ '  +  '{ \"name\": \"Antonio\", \"surname\": \"Cirillo\", \"age\": 21 }'  +  ', '  +  '{  \"name\": \"Giovanni\", \"surname\": \"Rapa\", \"age\": 21 }'  +  ' ]' 
	insertStudents = ""
	if statement[0] == "[" :
		insertStudents = json.loads(statement)
	else :
		insertStudents = json.loads("[" + statement + "]")
	
	nosql.insert_many(insertStudents)
	
	#__queue_service_client = __queue_service.get_queue_client('termination-"${function}"-"${id}"')
	#__queue_service_client.send_message('terminate')
main()